pkg/aws/awscontrolplane/validate_awscontrolplane.go

```diff
@@ -261,6 +315,28 @@ func contains(s []string, e string) bool {
 	}
 	return false
 }
+func countUniqueValues(s []string) int {
+	counter := make(map[string]int)
+	for _, a := range s {
+		counter[a]++
+	}
+	return len(counter)
+}
+func orderChanged(old []string, new []string) bool {
+	if len(old) > len(new) {
+		temp := old
+		old = new
+		new = temp
```

Comment:
Is there a chance we cause side effects of the slices to be modified by reference in another place where somebody cares about the state of the original slice? Slices are pointers AFAIK. So when you change `old` like that without working on a new memory representation on the stack somebody's expectations might fall short in another place. Just not sure if this is relevant in here.

Reply:
Good point, I should check it.